colfetcher: WIP on logical longest common prefix for a span

yuzefovich opened the pull request.

yuzefovich committed:
colfetcher: WIP on logical longest common prefix for a span

Multiple column families aren't supported.

Also, columns with composite key encodings that are in the index and
need to be decoded stop LLCP.

Release note: None

yuzefovich committed:
coldata: introduce new ExpandValue method

This commit introduces new `ExpandValue` method which allows for copying
a single value from a `Vec` into a range of indices of another `Vec` of
the same type.

TODO: add a unit test.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67726)
<!-- Reviewable:end -->

jordanlewis reviewed (commented):
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67726)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jordanlewis)



<!-- Sent from Reviewable.io -->

yuzefovich committed:
colbuilder: remove the check for index join on the interleaved index

Interleaved indexes are no more.

Release note: None

yuzefovich committed:
sql: remove the interleaved logic from fetchers

This commit removes the remaining bits of handling the interleaved
tables from both row fetcher and cFetcher.

Release note: None

yuzefovich committed:
row: simplify fetcher to only handle a single table

Previously, `row.Fetcher` could handle multiple tables at the same time.
This was needed to support the interleaved tables efficiently, but since
the interleaved tables are no more, we can simplify the fetcher now.

There are more things that can be removed from the fetcher about
handling of the interleaved tables, but I chose to defer them to a later
commit. This commit's main focus is simplifying the fetcher to work
under the assumption that it will be fetching only from a single table.

Release note: None

yuzefovich committed:
util/encoding: prevent a crash when decoding invalid decimal

Release note: None

yuzefovich committed:
WIP on multiple column families

As well as fixing some issues with tracing.

yuzefovich committed:
WIP fix the recent tracing refactor PR

yuzefovich commented:
cc @jordanlewis
